rust/chains/hyperlane-cosmos/src/providers/mod.rs

```diff
@@ -170,55 +219,27 @@ impl HyperlaneProvider for CosmosProvider {
         let tendermint_hash = Hash::from_bytes(Algorithm::Sha256, hash.as_bytes())
             .expect("transaction hash should be of correct size");
 
-        // TODO add proper error handling
         let response = self
             .rpc_client
             .tx(tendermint_hash, false)
             .await
-            .map_err(|_| ChainCommunicationError::from_other_str("generic error"))?;
+            .map_err(|e| {
+                ChainCommunicationError::from_other_str(&format!("error on RPC request: {}", e))
+            })?;
```

Comment:
I believe the error instance would already have the information conveyed by the string, so could be cleaner to just
```suggestion
            .map_err(Into::<HyperlaneCosmosError>::into)?;
```

Reply:
Thank you. Fixed.